### PR TITLE
Allow using device cam to take Pictures

### PR DESCRIPTION
**Pull request by andi34**
Fixes https://github.com/andreknieriem/photobooth/issues/210

Needs:
1. Access to Photobooth via HTTPS
2. `$config['previewFromCam'] = true;` and  `$config['previewCamTakesPic'] = true;`
3. Allow Photobooth in your browser to access your device camera

Who likes to give it a try can use my fork at https://github.com/andi34/photobooth which also contains different other open pull-requests and some other extras.

ezgif-7-b62cc744635a

### CODE REVIEW

**Comment by andi34** [comment]
There's a way to auto detect device cam height and width?

File: resources/js/core.js [hunk 1]
```diff
@@ -169,19 +171,22 @@ const photoBooth = (function () {
...
             style: photoStyle,
+            canvasimg: canvas.toDataURL('image/jpeg'),
         };
 
         if (photoStyle === 'collage') {
             data.file = currentCollageFile;
             data.collageNumber = nextCollageNumber;
         }
-
         jQuery.post('api/takePic.php', data).done(function (result) {
```
Comment (andi34) [inline resources/js/core.js]: keep this empty line

**Comment by andi34** [comment]
Anyone can help fixing the "jumping" preview at countdown?

**Comment by andi34** [comment]
Now everything works fine

**Comment by andi34** [comment]
Ok, there's one thing I now can't get fixed... The preview is zoomed on mobile access. ☹️

File: resources/js/core.js [hunk 2]
```diff
@@ -205,6 +212,7 @@ const photoBooth = (function () {
                 $('<p>').text(`${result.current + 1} / ${result.limit}`).appendTo('.loading');
                 $('#video--sensor').hide();
 
+
```
Comment (andi34) [inline resources/js/core.js]: Not needed this white line

File: resources/sass/style.scss [hunk 3]
```diff
@@ -344,15 +344,9 @@ hr.small {
   color: #fff;
 }
 
-#video--view #video--sensor {
```
Comment (andi34) [inline resources/sass/style.scss]: Forgot the comma between, with the comma the preview is visible outside the view.
Need to fix the styling somehow to not have the preview zoomed

**Comment by danielva92** [comment]
Hi, thanks for your work
What do i have to do to Access Photobooth via HTTPS?
Greets

**Comment by andi34** [comment]
> Hi, thanks for your work
> What do i have to do to Access Photobooth via HTTPS?
> Greets

Take a look at Google search for "self signed ssl certificate", I am still new to this which why I can't give a proper answer. Take a look at the issue post here https://github.com/andreknieriem/photobooth/issues/202 if you like, I've posted there how I've got it working using Apache Webserver.

**Comment by sualko** [comment]
> What do i have to do to Access Photobooth via HTTPS?

Because your browser wants to protect you from someone interfering with your camera/mic and so on. If you access photobooth via localhost, http should be enough, see this [mailing list post](https://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features) for details. If you like you can also create an [exception](https://medium.com/@Carmichaelize/enabling-the-microphone-camera-in-chrome-for-local-unsecure-origins-9c90c3149339).

**Pull request merged**